### DISCUSSION TITLE
Make GraphQL system-added suffixes/prefixes more arbitrary

### DISCUSSION THREAD

**Discussion by renomarx** [directus/directus#13558] [Feature Requests]
### Preflight Checklist

- [X] I have completed all [Troubleshooting Steps](https://docs.directus.io/getting-started/support/#troubleshooting-steps).
- [X] I'm on [the latest version of Directus](https://github.com/directus/directus/releases).
- [X] There's [no other issue](https://github.com/directus/directus/issues?q=is%3Aissue) that already describes my problem.

### Describe the Bug

We have 2 collections:
- 1 named `some_table`
- The other named `some_table_filter`

When we try to reach Directus GraphQL API, even querying other collections, we have an error:
```
{"errors":[{"message":"Schema must contain uniquely named types but contains multiple types named \"some_table_filter\".","extensions":{"code":"INTERNAL_SERVER_ERROR"}}]}
```

When I check the code, I see something that could explain that:
https://github.com/directus/directus/blob/main/api/src/services/graphql.ts#L746

Does that mean that we can't currently have this kind of data model if we want to use the GraphQL API ?

### To Reproduce

`npm init directus-project example-project`

Create 2 collections:
- 1 named `some_table`
- 1 named `some_table_filter`
And make them public (for reading at least)

Then:
```
curl --request POST \
  --url http://localhost:8055/graphql \
  --header 'Content-Type: application/json' \
  --data '{"query":"query {\n\tsome_table {\n\t\tid\n\t}\n}\n"}'
```

And check Directus logs:

### Errors Shown

```
17:46:42 🚨 Schema must contain uniquely named types but contains multiple types named "some_table_filter". 
Error: Schema must contain uniquely named types but contains multiple types named "some_table_filter".
    at new GraphQLSchema (/home/reno/work/directus-test/node_modules/graphql/type/schema.js:194:15)
    at SchemaComposer.buildSchema (/home/reno/work/directus-test/node_modules/graphql-compose/lib/SchemaComposer.js:128:16)
    at GraphQLService.getSchema (/home/reno/work/directus-test/node_modules/directus/dist/services/graphql.js:230:31)
    at GraphQLService.execute (/home/reno/work/directus-test/node_modules/directus/dist/services/graphql.js:91:29)
    at /home/reno/work/directus-test/node_modules/directus/dist/controllers/graphql.js:27:40
    at scoped (/home/reno/work/directus-test/node_modules/directus/dist/utils/async-handler.js:5:60)
    at Layer.handle [as handle_request] (/home/reno/work/directus-test/node_modules/express/lib/router/layer.js:95:5)
    at trim_prefix (/home/reno/work/directus-test/node_modules/express/lib/router/index.js:323:13)
    at /home/reno/work/directus-test/node_modules/express/lib/router/index.js:284:7
    at Function.process_params (/home/reno/work/directus-test/node_modules/express/lib/router/index.js:341:12)
17:46:42 ✨ request errored POST 500 /graphql 43ms

```

### What version of Directus are you using?

9.6.0

### What version of Node.js are you using?

16.14.0

### What database are you using?

MySQL/SQLite

### What browser are you using?

Firefox

### What operating system are you using?

Linux

### How are you deploying Directus?

Docker / running locally

**Comment by rijkvanzanten** [comment]
That's currently correct. GraphQL doesn't really have a good way to separate names other than using suffixes / prefixes. In this case, the suffix we use for filters is `_filter`. That being said, we could update Directus' internal suffixes to be more arbitrary to prevent this particular edge case. (although that'd hurt the auto-generation of specs, as the names would become very mangled / annoying)

**Comment by rijkvanzanten** [comment]
Heya! Thank you for taking the time to submit this request! 

It has been over **90 days**, and this discussion has not received at least **15 votes** from the community. This means that we don't feel like there's enough community interest to warrant further R&D into this topic at this time. :ice_cube:

This request will now be closed to keep our discussions tidy. Please reach out if you have any questions!

_For more information, see our [Feature Request Process](https://docs.directus.io/contributing/feature-request-process.html)._
